ports/servo/browser.rs

```diff
@@ -106,37 +106,40 @@ impl Browser {
                     }
                 }
             }
-            (CMD_OR_CONTROL, Some('q'), _) => {
+            (CMD_OR_CONTROL, Some('q'), _) => if state == KeyState::Pressed {
                 self.event_queue.push(WindowEvent::Quit);
             }
-            (_, Some('3'), _) if mods ^ KeyModifiers::CONTROL == KeyModifiers::SHIFT => {
+            (_, Some('3'), _) =>
+            if state == KeyState::Pressed && mods ^ KeyModifiers::CONTROL == KeyModifiers::SHIFT {
```

Comment:
Because you moved the modifier check after `=>`, it means that just pressing "3" will never make it to Servo.

Keep the modifier check before `=>`.